alma/lib/Model/PaymentData.php

```diff
@@ -175,7 +176,26 @@ public static function dataFromCart($cart, $context, $feePlans, $forPayment = fa
             $customerData['business_name'] = $billingAddress->company;
         }
 
+        $carrierHelper = new CarrierHelper($context);
+        $cartHelper = new CartHelper($context);
+
+        $websiteCustomerDetails = [
+            'new_customer' => self::getNewCustomer($customer->id),
+            'is_guest' => (bool) $customer->is_guest,
+            'created' => strtotime($customer->date_add),
+            'current_order' => [
+                'purchase_amount' => almaPriceToCents($purchaseAmount),
+                'payment_method' => 'alma',
+                'shipping_method' => $carrierHelper->getNameCarrierById($cart->id_carrier),
+                'items' => CartData::cartItems($cart),
+            ],
+            'previous_orders' => [
+                $cartHelper->previousCartOrdered($customer->id),
+            ],
+        ];
+
         $dataPayment = [
+            'website_customer_details' => $websiteCustomerDetails,
```

Comment:
you could add a `buildWebsiteCustomerDetails()` method instead of building it into the already big method `dataFromCart`

Then, you should test this method to ensure the format will be as the format expected.

Reply:
I thought with the refacto that will be better, because is very hard to make test with static of private function